addons/html_builder/static/src/builder/options/device_visibility_option.js

```diff
@@ -0,0 +1,97 @@
+import { registry } from "@web/core/registry";
+import { Plugin } from "@html_editor/plugin";
+class DeviceVisibilityOptionPlugin extends Plugin {
+    static id = "DeviceVisibilityOption";
+    websiteService = this.services.website;
+    selector = "section .row > div";
+    resources = {
+        builder_options: {
+            template: "html_builder.DeviceVisibilityOption",
+            selector: this.selector,
+        },
+        builder_actions: this.getActions(),
+        clean_for_save_handlers: this.cleanForSave.bind(this),
+    };
+    cleanForSave({ root }) {
+        const editingEls = root.querySelectorAll(this.selector);
+        for (const editingEl of editingEls) {
+            const invisibleClass = editingEl.classList.contains("o_snippet_desktop_invisible")
+                ? "no_desktop"
+                : "" || editingEl.classList.contains("o_snippet_desktop_invisible")
+                ? "no_mobile"
+                : "";
+            if (invisibleClass) {
+                this.getActions().toggleDeviceVisibility.apply({
+                    editingElement: editingEl,
+                    param: invisibleClass,
+                });
+                editingEl.classList.remove("o_snippet_override_invisible");
+            }
+        }
+    }
+    getActions() {
+        return {
+            toggleDeviceVisibility: {
+                apply: ({ editingElement, param }) => {
+                    // Clean first as the widget is not part of a group
+                    this.clean(editingElement);
+                    const style = getComputedStyle(editingElement);
+                    if (param === "no_desktop") {
+                        editingElement.classList.add("d-lg-none", "o_snippet_desktop_invisible");
+                    } else if (param === "no_mobile") {
+                        editingElement.classList.add(
+                            `d-lg-${style["display"]}`,
+                            "d-none",
+                            "o_snippet_mobile_invisible"
+                        );
+                    }
+
+                    // Update invisible elements
+                    const isMobile = this.websiteService.context.isMobile;
+                    const show = param !== (isMobile ? "no_mobile" : "no_desktop");
+                    if (!show) {
+                        editingElement.dataset.invisible = "1";
+                    }
```

Comment:
I think that this part and the same one in the `invisible_elements_panel.js` should be extracted in a plugin or something with a resource making these changes
It will be similar to the trigger to `snippet_option_visibility_update` and will allow to group the logic at the same place :slightly_smiling_face: